packages/e2e-test/tests/light-client-rpc.e2e.test.ts

```diff
@@ -0,0 +1,159 @@
+import anyTest, { TestInterface } from "ava";
+import { asyncSleep, generateHDAccount, HDAccount } from "../src/utils";
+import { CKB_RPC_URL, LIGHT_CLIENT_RPC_URL } from "../src/constants";
+import { E2EProvider } from "../src/e2eProvider";
+import { join } from "path";
+import { FileFaucetQueue } from "../src/faucetQueue";
+import { Cell } from "@ckb-lumos/base";
+import { Config } from "@ckb-lumos/config-manager";
+import { RPC } from "@ckb-lumos/rpc";
+import {
+  CellCollector,
+  Indexer,
+  LightClientRPC,
+  TerminableCellAdapter,
+} from "@ckb-lumos/ckb-indexer";
+import { isEqual } from "lodash";
+
+interface TestContext {
+  accounts: Record<
+    string,
+    HDAccount & {
+      cells: Cell[];
+    }
+  >;
+  config: Config;
+}
+const test = anyTest as TestInterface<TestContext>;
+
+const rpc = new RPC(CKB_RPC_URL);
+const indexer = new Indexer(CKB_RPC_URL);
+const lightClientRPC = new LightClientRPC(LIGHT_CLIENT_RPC_URL);
+const faucetQueue = new FileFaucetQueue(join(__dirname, "../tmp/"));
+const e2eProvider = new E2EProvider({ indexer, rpc, faucetQueue: faucetQueue });
+
+async function waitLightClientPrepared(lightClientRpc: LightClientRPC) {
+  while (true) {
+    try {
+      await lightClientRpc.getTipHeader();
+      return;
+    } catch (err) {
+      asyncSleep(1000);
+    }
+  }
+}
+
+async function waitLightClientSync(
+  lightClientRpc: LightClientRPC,
+  targetBlockNumber: number
+) {
+  while (true) {
+    const scripts = await lightClientRpc.getScripts();
+
+    const done = !scripts
+      .map((script) => parseInt(script.blockNumber))
+      .some((n) => n < targetBlockNumber);
+
+    if (done) {
+      return;
+    }
+
+    await asyncSleep(3000);
+  }
+}
+
+test.before(async (t) => {
+  const config = await e2eProvider.loadLocalConfig();
+  const alice = generateHDAccount();
```

Comment:
I found that `generateHDAccount` is not related to HD wallet, we need to rename it to `randomSecp256k1Lock`